tests/end_to_end/netsh_test.cpp

```diff
@@ -779,6 +779,86 @@ TEST_CASE("xdp interface parameter", "[netsh][programs]")
+    REQUIRE(
+        output == "\n"
+                  "    ID  Pins  Links  Mode       Type           Name\n"
+                  "======  ====  =====  =========  =============  ====================\n"
+                  "196614     1      0  JIT        xdp            DropPacket\n");
+
+    // Re-attach the program with interface index parameter.
+    output = _run_netsh_command(handle_ebpf_set_program, L"196614", nullptr, L"interface=1", &result);
+    REQUIRE(output == "");
+    REQUIRE(result == ERROR_OKAY);
+    output = _run_netsh_command(handle_ebpf_delete_program, L"196614", nullptr, nullptr, &result);
+    REQUIRE(result == NO_ERROR);
+
+    ebpf_epoch_flush();
+}
+
+TEST_CASE("cgroup_sock_addr compartment parameter", "[netsh][programs]")
+{
+    _test_helper_netsh test_helper;
+
+    // Load a program pinned.
+    int result;
+
+    // Load program with pinpath and loopback interface alias.
+    std::string output = run_netsh_command_with_args(
+        handle_ebpf_add_program, &result, 4, L"cgroup_sock_addr.o", L"cgroup/connect4", L"mypinpath", L"compartment=1");
+    REQUIRE(strcmp(output.c_str(), "Loaded with ID 196609\n") == 0);
+    REQUIRE(result == NO_ERROR);
+    output = _run_netsh_command(handle_ebpf_delete_program, L"196609", nullptr, nullptr, &result);
+    REQUIRE(result == NO_ERROR);
+    REQUIRE(output == "Unpinned 196609 from mypinpath\n");
+    verify_no_programs_exist();
+
+    // Load program with pinpath and loopback interface name.
+    output = run_netsh_command_with_args(
+        handle_ebpf_add_program, &result, 4, L"droppacket.o", L"xdp", L"mypinpath", L"loopback_0");
+    REQUIRE(strcmp(output.c_str(), "Loaded with ID 196610\n") == 0);
+    REQUIRE(result == NO_ERROR);
+    output = _run_netsh_command(handle_ebpf_delete_program, L"196610", nullptr, nullptr, &result);
+    REQUIRE(result == NO_ERROR);
+    REQUIRE(output == "Unpinned 196610 from mypinpath\n");
+    verify_no_programs_exist();
+
+    // Load program with loopback interface index.
+    output = _run_netsh_command(handle_ebpf_add_program, L"droppacket.o", L"xdp", L"interface=1", &result);
+    REQUIRE(strcmp(output.c_str(), "Loaded with ID 196611\n") == 0);
+    REQUIRE(result == NO_ERROR);
+    output = _run_netsh_command(handle_ebpf_delete_program, L"196611", nullptr, nullptr, &result);
+    REQUIRE(result == NO_ERROR);
+    REQUIRE(output == "Unpinned 196611 from DropPacket\n");
+    verify_no_programs_exist();
+
+    // (Negative) Load program with incorrect interface name.
+    output = _run_netsh_command(handle_ebpf_add_program, L"droppacket.o", L"xdp", L"interface=foo", &result);
+    REQUIRE(strcmp(output.c_str(), "Interface parameter is invalid.\n") == 0);
+    REQUIRE(result == ERROR_SUPPRESS_OUTPUT);
+    verify_no_programs_exist();
+
+    // (Negative) Load program with program type that does not support interface parameter.
+    output = _run_netsh_command(handle_ebpf_add_program, L"bindmonitor.o", L"bind", L"interface=1", &result);
```

Comment:
Should we add a similar negative test for program type which does not support compartment ID?

Reply:
Updated